src/rules/semicolonRule.ts

```diff
@@ -222,8 +222,19 @@ class SemicolonWalker extends Lint.AbstractWalker<Options> {
                 case ts.SyntaxKind.RegularExpressionLiteral:
                     break;
                 default:
-                    this.reportUnnecessary(node.end - 1);
+                    if (!this.isFollowedByStatement(node)) {
+                        this.reportUnnecessary(node.end - 1);
+                    }
             }
         }
     }
+
+    private isFollowedByStatement(node: ts.Statement): boolean {
+        const nextStatement = utils.getNextStatement(node);
+        if (nextStatement === undefined) {
+            return false;
+        }
+        return ts.getLineAndCharacterOfPosition(this.sourceFile, node.end).line
```

Comment:
> This should be a utility somewhere

Sure, will do that later on.

> The current method would be tricked by a multi-line statement followed by another

The rule handles your example as expected. I added it as a test to make sure this wont regress.